recipes/ckanbatch-setup.rb

```diff
@@ -40,11 +40,35 @@
 	EOS
 end
 
-# Managed processes sometimes don't shut down properly on daemon stop,
-# leaving them 'orphaned' and resulting in duplicates.
-# Work around by issuing a stop command to the children first.
-execute "Stop children on supervisord stop" do
-	command <<-'SED'.strip + " /etc/init.d/supervisord"
-		sed -i 's/^\(\s*\)\(killproc\)/\1timeout 10s supervisorctl stop all || echo "WARNING: Unable to stop managed process(es) - check for orphans"; \2/'
-	SED
+if ::File.exist? "/etc/init.d/supervisord" then
+	# Managed processes sometimes don't shut down properly on daemon stop,
+	# leaving them 'orphaned' and resulting in duplicates.
+	# Work around by issuing a stop command to the children first.
+	execute "Stop children on supervisord stop" do
+		command <<-'SED'.strip + " /etc/init.d/supervisord"
+			sed -i 's/^\(\s*\)\(killproc\)/\1timeout 10s supervisorctl stop all || echo "WARNING: Unable to stop managed process(es) - check for orphans"; \2/'
+		SED
+	end
+else
+	systemd_unit "supervisord.service" do
```

Comment:
do we have systemd in linux1? do we have a way to be 'cross-compatible'

Reply:
I checked and `/etc/systemd` exists on Amazon Linux 1, but the daemon isn't actually running.

This is backwards compatible because Supervisord on Amazon Linux 2 will not create `/etc/init.d/supervisord`. So, if initd config exists, use that; else configure systemd.